infrastructure/provider/flight_reserve.go

```diff
@@ -0,0 +1,42 @@
+package provider
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/cyneptic/letsgo/internal/core/entities"
+)
+
+type SortFilterProviderClient struct {
+	client  *http.Client
+	address string
+}
+
+func NewSortFilterProviderClient(address string) *SortFilterProviderClient {
+	tr := &http.Transport{}
+	cl := &http.Client{Transport: tr}
+
+	return &SortFilterProviderClient{
+		client:  cl,
+		address: address,
+	}
+}
+
+func (pc *SortFilterProviderClient) RequestFlight() ([]entities.Flight, error) {
+	resp, err := pc.client.Get(pc.address + "/flights?source=tehran&destination=sari&departing=2023-07-29")
```

Comment:
this is a hardcoded request, the client must decide what flight they are looking for.